### PR TITLE
infra/gen-view: Fix links solving

### PR DESCRIPTION
**Pull request by Alex-deVis**
Solve links for tasks and guides, not only for reading. This implies that all reading content in `config.yaml` should be prefixed with 'reading/'.

# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [x] Tested your changes against relevant architectures and platforms;
- [x] Updated relevant documentation (if needed).

### CODE REVIEW

**Review by teodutu — CHANGES_REQUESTED** [review]
This breaks links in Markdown files (in the repo itself) which makes it difficult to see connections if you're looking at a single reading / task item. Can't you do it while preserving the Markdown links?

**Comment by Alex-deVis** [comment]
> This breaks links in Markdown files (in the repo itself) which makes it difficult to see connections if you're looking at a single reading / task item. Can't you do it while preserving the Markdown links?

Did my best to preserve the links, however, it would be too much work to do it properly for tasks. Currently, I am copying the task statement from `tasks/<task-name>/README.md` to `.view/<task-name>.md` to avoid name conflicts. Considering that we are planning to change the structure soon, I think we can live with this.

There is one more problem related to relative links to the same subdirectory (e.g. `../some-other-guide.md`). Solving them requires the script to be aware of the current file's location and I want to keep the script lean until we fully commit to a structure.

So in short, we have to make these concessions to have valid links in the generated view, but they are solvable if the structure stays like this.

**Comment by teodutu** [comment]
> Considering that we are planning to change the structure soon, I think we can live with this.

Ok, fair enough.

**Review by teodutu — APPROVED** [review]
Ship it
